crates/wasmi/src/store.rs

```diff
@@ -67,6 +70,28 @@ impl StoreIdx {
 /// A stored entity.
 pub type Stored<Idx> = GuardedEntity<StoreIdx, Idx>;
 
+pub struct ResourceLimiterRef<'a>(Option<&'a mut (dyn crate::ResourceLimiter)>);
+impl<'a> core::fmt::Debug for ResourceLimiterRef<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "ResourceLimiterRef(...)")
+    }
+}
+
+impl<'a> ResourceLimiterRef<'a> {
+    pub fn as_resource_limiter(&mut self) -> &mut Option<&'a mut dyn crate::ResourceLimiter> {
+        &mut self.0
+    }
+}
+
+struct ResourceLimiterQuery<T>(
```

Comment:
I'd love to see a (short) doc comment explaining the rational behind the existence of this type for our future selves. :)

Reply:
Done.